src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs

```diff
@@ -644,6 +657,18 @@ private static bool InRange(int ch, int start, int end)
         /// <returns>True if the input buffer was fully encoded into the output buffer, otherwise false.</returns>
         public static unsafe bool TryEncode(ReadOnlySpan<char> utf16, Span<byte> utf8, out int charactersConsumed, out int bytesWritten)
         {
+            if (utf16.IsEmpty)
```

Comment:
I think it would be better to fix the loop below to subtract the pointers instead - like `while (pEnd - pSrc < 13) instead of this upfront check.